src/main/kotlin/com/hibob/academy/resource/PetResource.kt

```diff
@@ -61,4 +61,18 @@ import org.springframework.web.bind.annotation.RequestBody
             petService.deletePet(petId, companyId)
             return Response.status(Response.Status.OK).entity("Deleted").build()
         }
+
+        @GET
+        @Path("{ownerId}/petsOfOwner")
+        fun getAllPets(@PathParam("ownerId") ownerId: Long, @QueryParam("companyId") companyId: Long): Response {
```

Comment:
why companyId is QueryParam and not PathParam?

Reply:
because all functions get companyId even when it's not really relevant i thought of doing it that way. Is it more correct to use PathParam?

Reply:
The choice between query parameters and path parameters depends on the use case:

Use Path Parameters when:
The value is a unique identifier for a resource.
It’s essential to the endpoint's structure and directly represents the resource being accessed.
Example: /pets/{id} or /users/{username} for accessing a specific pet or user.
Use Query Parameters when:
The value is optional or used for filtering, sorting, or pagination.
It modifies the results or specifies additional criteria without changing the endpoint's structure.
Example: /pets?type=dog&age=2 for filtering pets by type and age.
Summary:
Path parameters for identifying resources.
Query parameters for modifying or filtering the results of a resource.

Reply:
thanks for the clarification, i'll change it!

Reply:
🤔

Reply:
weird, I changed it